mokume/io/feature.py

```
@@ -0,0 +1,605 @@
+                COALESCE(s.sdrf_sample_accession, {sa_fallback}) as sample_accession,
+                p.channel,
+                p.intensity,
+                p.run,
+                COALESCE(s.sdrf_condition, {sa_fallback}) as condition,
+                COALESCE(CAST(s.sdrf_biological_replicate AS INTEGER), 1) as biological_replicate,
+                COALESCE(CAST(s.sdrf_fraction AS VARCHAR), '1') as fraction,
+                split_part(COALESCE(s.sdrf_sample_accession, {sa_fallback}), '_', 1) as mixture
+            FROM parquet_db_unnested p
+            LEFT JOIN sdrf_mapping s
+                {join_clause}
+        """)
```

Ruff (0.15.5)

[error] Possible SQL injection vector through string-based query construction

(S608)